pyscf/neo/hf.py

```diff
@@ -378,17 +378,119 @@ def _build_eri(mol1, mol2, cart):
                                              size1, size1 + size2),
                                  aosym='s4')
 
+def cholesky_eneri(mol_n, auxmol_e, int3c, int2c, max_memory,
+                   aosym='s2ij', decompose_j2c='cd'):
+    from pyscf.df.incore import _eig_decompose
+    from pyscf.df.outcore import _guess_shell_ranges
+    j2c_e = auxmol_e.intor(int2c, hermi=1)
+    if decompose_j2c == 'eig':
+        low = _eig_decompose(auxmol_e, j2c_e)
+    else:
+        try:
+            low = scipy.linalg.cholesky(j2c_e, lower=True)
+            decompose_j2c = 'cd'
+        except scipy.linalg.LinAlgError:
+            low = _eig_decompose(auxmol_e, j2c_e)
```

Comment:
Same issue here. Maybe you can just let the function crash if it is not 'cd' to see if the larger error comes from using incorrect _eig_decompose.